### ISSUE TITLE
[Feature Request] Support for Dynamic Path Resolution

### ISSUE THREAD

**Issue by hotic** [dlevs/vscode-regex-robin#8]
I'm currently utilizing your Regex Robin VS Code extension and I'm thrilled with the functionality it provides.

I'm working on a project where I have YAML files (e.g., datasource.yml) scattered across various subdirectories within my workspace, and I often need to navigate to these files directly from the editor. I attempted to use Regex Robin to create navigable links by defining rules in my settings like so:

```
"regexrobin.rules": [
  {
    "regex": "(\\b\\w+\\.yml)",
    "editor": [
      {
        "link": "${workspaceFolder}/$1",
      }
    ]
  }
``` 

This setup works well if the YAML files are located at the root of the workspace. But it fails to find files in subdirectories because the rule is not dynamically resolving the file paths. possibly requiring a recursive search mechanism or the like. it would profoundly enhance the plugin's utility for projects with a more complex directory structure.

**Comment by dlevs** [comment]
Glad you find the plugin useful. :)

Have you tried this?

```diff
  "regexrobin.rules": [
    {
      "regex": "(\\b\\w+\\.yml)",
      "editor": [
        {
-         "link": "${workspaceFolder}/$1",
+         "link": "${fileDirname}/$1",
        }
      ]
    }
```

That works when the `datasource.yml` is in the same directory as the file referencing it.

If you need something more complex, you can also put arbitrary JS inside the `${}` brackets, and `fileDirname` is just a normal string value:

```json
{
  "link": "${fileDirname.replaceAll('a', 'b')}/$1",
}
```

If you need something more advanced, maybe share a minimal zip file that represents your project so I know more what you're looking for.

**Comment by hotic** [comment]
Of course! I have a detailed example here. 

```
Data
 |- module 
   |- datasource
      |- datasource_1.yml
      |- datasource_2.yml
      |- datasource_3.yml
   |- datasource_old
      |- datasource.yml
   |- test.yml 
 |- data.db
 |- dataconfig.yml 
``` 
Within `test.yml`, I have a section like this:
```
default_0:
  datasource:
    - 'datasource.yml'
    - 'datasource_1.yml'
    - 'datasource_2.yml'
    - 'datasource_3.yml'
  config:
    - 'dataconfig.yml'
``` 
My goal is to be able to click on the 'datasource.yml' reference within `test.yml` and have it open the `datasource.yml` file located at `Data/module/datasource_old/datasource.yml,` or any `.yml` located in the directory of Data folder. If there are multiple matches, defaulting to the first occurrence would suit my needs.

Thank you for considering this enhancement, such a feature would significantly aid navigation and efficiency.

**Comment by dlevs** [comment]
Understood! That would be a cool feature. Basically, it would be fuzzy link resolution. I'm up for implementing it, but I won't get a chance any time soon.

Similar functionality exists in another plugin I use: [Markdown Notes](https://marketplace.visualstudio.com/items?itemName=kortina.vscode-markdown-notes). It won't work for your use case (requires the link to be wrapped in `[[brackets]]`), but there might be similar extensions out there.

For future reference, in case I pick this up later, I looked into how Markdown Notes implements this. I'd copy the following implementation details:
- **Fuzzy file resolution.** Uses `vscode.workspace.findFiles()` to get matches.
- **Creating files on click if they don't exist.** This is a "Markdown Notes" feature I use all the time. If this existed in Regex Robin, I'd be able to uninstall Markdown Notes. This would involve replacing `vscode.DocumentLinkProvider` with `vscode.DefinitionProvider`, so that a callback is called when clicking the link.

So in short, I like this idea. Thanks for the suggestion. I might implement it, but it's unlikely to be soon.
